.github/workflows/terraform-observe_release.yaml

```
@@ -24,6 +24,11 @@ jobs:
       contents: write
       pull-requests: write
     steps:
+      - name: Check branch
+        if: github.ref_name != github.event.repository.default_branch
```

```suggestion
        if: github.ref_type != 'branch' || github.ref_name != github.event.repository.default_branch
```

Otherwise this condition won't trigger given `git tag main`.